### DISCUSSION TITLE
How to generate a certificate with a new custom Micro  service with or without batch?

### DISCUSSION THREAD

**Discussion by nidhikachhawa24** [Sunbird-Ed/Community#739] [Enhancement request]
Hello @maheshkumargangula, @Krishnaj20, @rajeevsathish,

We have a use case where we need to issue the certificate to user for other contents/scenario than course completion
We want to write new Micro Service for it which will directly issue a certificate to users.

**_Analysis:_**
We have found out that following flow is happening in current system,
Flink job calls -> activity aggregate updater -> it will keep a track of users  ==> **this updates database to mark content consumption status as completed**
Once it matches the criteria -> an event is called to another flink job certificate generator pre-processor -> this again generates event to collection certificate generator ==> **this we need to skip as we do want to create new fling job**
-> again an event is generated to RC block -> certificate api and certificate sign (certificate is generated)

But we don't want fling job dependency to call certificate issue/generate API calls. we want it to be called directly through custom micro service.

**Want to understand how following points work:**
Is it possible to create or issue a certificate with API calls
Can we directly call RC block APis to generate certificate through  certificateapi and certificatesign?

cc: @coolbung, @vinukumar-vs @paritshivani @komalm @manishakapadnis

**Comment by HarishGangula** [comment]
@nidhikachhawa24  can you ask this question in https://github.com/orgs/Sunbird-Lern/discussions ?

**Reply by paritshivani** [reply]
Added ticket in lern. thank you:
[https://github.com/orgs/Sunbird-Lern/discussions/206](https://github.com/orgs/Sunbird-Lern/discussions/206)

**Comment by HarishGangula** [comment]
Closing this since it is moved to lern
